Test: Harden lint-filenames.sh

wodry opened the pull request:
- This fixes that only files with lower case file name suffix where found before, which contradicted the Regex to find uppercase characters in file names (including suffixes I guess).
- `--full-name` switch was added to git ls-files, to define that the found file always includes it's full path in the git project.
- since we know now that the file name includes the full path, we can harden the Regex to exclude the  secp256k1 and univalue sub folders.
- use backslash line break to make code easier to read and avoid too long line.

Commit:
Test: Harden lint-filenames.sh

practicalswift commented on test/lint/lint-filenames.sh:
```diff
@@ -10,7 +10,11 @@
+OUTPUT=$(git ls-files --full-name -- "*.[cC][pP][pP]" "*.[hH]" "*.[pP][yY]" "*.[sS][hH]" \
+| grep -vE '^[a-z0-9_./-]+$' \
+| grep -vE '^src/(secp256k1|univalue)/' \
+)
```
Nit: Please format it like this to make it consistent with the rest of the lint shell scripts:

```
OUTPUT=$(git ls-files --full-name -- "*.[cC][pP][pP]" "*.[hH]" "*.[pP][yY]" "*.[sS][hH]" | \
    grep -vE '^[a-z0-9_./-]+$' | \
    grep -vE '^src/(secp256k1|univalue)/')
```

practicalswift commented:
utACK 8fd05107ee5c188c9ab59b0c1b3f5698f8075c8d modulo nits

Luckily:

```
$ git log -u | grep -E '^\+\+\+ b/' | grep -E '\.(CPP|H|PY|SH)$'
$
```

wodry commented:
Changed coding style, allthough I think to have the closing bracket on it's own line would be clearer to read and less error prone if a line should be attached, also better diff reading. But that's no discussion for a single PR, so changed according to the existing scripting stle as requested. Rebased and squashed.

practicalswift commented:
ACK 927e1150bc207181f7d5b0948e87d04b737a1c27

wodry commented:
Yes, only lowercase suffixes, here the Regex that also would find mixed case suffixes:
```
git log -u | grep -E '^\+\+\+ b/' | grep -iE '\.(cpp|h|py|sh)$' | grep -cvE '\.(cpp|h|py|sh)$'
0
```

ken2812221 reviewed (approved):
utACK 927e115
